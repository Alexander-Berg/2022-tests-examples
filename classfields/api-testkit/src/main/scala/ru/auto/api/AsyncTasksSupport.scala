package ru.auto.api

import io.prometheus.client.Counter

trait AsyncTasksSupport {

  implicit val asyncTasksCounter: Counter = Counter
    .build("test_async_counter", "Async tasks counter for tests")
    .labelNames("name", "error")
    .create()
}
