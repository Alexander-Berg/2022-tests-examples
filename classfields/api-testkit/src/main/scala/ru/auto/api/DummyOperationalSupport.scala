package ru.auto.api

import java.util.function.Predicate
import io.prometheus.client.{Collector, CollectorRegistry}
import ru.yandex.vertis.ops.MetricsSupport
import ru.yandex.vertis.ops.codahale.CodahaleRegistry
import ru.yandex.vertis.ops.prometheus.PrometheusRegistry

trait DummyOperationalSupport {

  implicit val metricsSupport: MetricsSupport = new MetricsSupport {
    implicit override def codahaleRegistry: CodahaleRegistry = null

    implicit override def prometheusRegistry: PrometheusRegistry = prometheusRegistryDummy
  }

  implicit val prometheusRegistryDummy: PrometheusRegistry = new PrometheusRegistry {
    override def register[C <: Collector](c: C): C = c

    override def asCollectorRegistry(): CollectorRegistry = new CollectorRegistry()

    override def unregister(p: Predicate[Collector]): Unit = ()
  }
}
