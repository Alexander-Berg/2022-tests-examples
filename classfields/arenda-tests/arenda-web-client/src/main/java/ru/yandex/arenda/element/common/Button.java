package ru.yandex.arenda.element.common;

import io.qameta.atlas.webdriver.AtlasWebElement;
import io.qameta.atlas.webdriver.extension.FindBy;
import io.qameta.atlas.webdriver.extension.Name;
import io.qameta.atlas.webdriver.extension.Param;

public interface Button extends AtlasWebElement {

    @Name("Кнопка «{{ value }}»")
    @FindBy(".//button[contains(.,'{{ value }}')]")
    ArendaElement button(@Param("value") String value);

    @Name("Кнопка «{{ value }}»")
    @FindBy(".//button[.='{{ value }}']")
    ArendaElement exactButton(@Param("value") String value);

    @Name("Кнопка «{{ value }}»")
    @FindBy(".//a[contains(.,'{{ value }}')]")
    ArendaElement buttonRequest(@Param("value") String value);

    @Name("Кнопка")
    @FindBy(".//button")
    ArendaElement button();
}
