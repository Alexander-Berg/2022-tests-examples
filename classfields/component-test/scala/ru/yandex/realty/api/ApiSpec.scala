package ru.yandex.realty.api

import akka.http.scaladsl.model.StatusCodes
import org.junit.runner.RunWith
import org.scalatestplus.junit.JUnitRunner
import ru.yandex.realty.AbstractBnbSearcherComponentTestSpec

import scala.language.implicitConversions

@RunWith(classOf[JUnitRunner])
class ApiSpec extends AbstractBnbSearcherComponentTestSpec {

  "GET /ping" should {

    "return OK" in {

      Get("/ping") ~>
        route ~>
        check {
          status should be(StatusCodes.OK)
        }

    }

  }

}
