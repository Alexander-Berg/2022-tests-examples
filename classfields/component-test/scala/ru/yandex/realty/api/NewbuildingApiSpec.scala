package ru.yandex.realty.api

import akka.http.scaladsl.model.StatusCodes
import org.junit.runner.RunWith
import org.scalatestplus.junit.JUnitRunner
import ru.yandex.realty.AbstractBnbSearcherComponentTestSpec
import ru.yandex.realty.api.ProtoResponse.NewbuildingContactsResponse
import ru.yandex.realty.componenttest.data.campaigns.Campaign_56576
import ru.yandex.realty.componenttest.data.sites.Site_57547
import ru.yandex.realty.model.phone.RealtyPhoneTags.MapsMobileTagName

import scala.language.implicitConversions

@RunWith(classOf[JUnitRunner])
class NewbuildingApiSpec extends AbstractBnbSearcherComponentTestSpec {

  import NewbuildingApiSpec._

  "GET /newbuilding/{newbuildingId}/contacts" should {

    "return newbuilding contacts" in {

      val newbuildingId = Site_57547.Id
      val expectedRedirectObjectId = newbuildingRedirectObjectId(newbuildingId)
      val phone = Campaign_56576.Proto.getRedirects(0).getSource

      stubTeleponyGetOrCreateOk(objectId = expectedRedirectObjectId, phone)
      stubTeleponyDeleteOk(objectId = expectedRedirectObjectId)

      Get(s"/newbuilding/$newbuildingId/contacts") ~>
        route ~>
        check {
          status should be(StatusCodes.OK)
          val response = entityAs[NewbuildingContactsResponse]
          response.getResponse.getSalesDepartmentsCount === 1
        }

    }

    "return newbuilding contacts when tag is provided" in {

      val newbuildingId = Site_57547.Id
      val expectedRedirectObjectId = newbuildingRedirectObjectId(newbuildingId)

      stubTeleponyDeleteOk(objectId = expectedRedirectObjectId)

      Get(s"/newbuilding/$newbuildingId/contacts?tag=$MapsMobileTagName") ~>
        route ~>
        check {
          status should be(StatusCodes.OK)
          val response = entityAs[NewbuildingContactsResponse]
          response.getResponse.getSalesDepartmentsCount === 1
        }

    }

    s"return status=${StatusCodes.NotFound} when newbuilding does not exist" in {

      val nonExistingNewbuildingId = 10101010

      Get(s"/newbuilding/$nonExistingNewbuildingId/contacts") ~>
        route ~>
        check {
          status should be(StatusCodes.NotFound)
        }

    }

    "return newbuilding contacts when failed to remove old redirects" in {

      val newbuildingId = Site_57547.Id
      val expectedRedirectObjectId = newbuildingRedirectObjectId(newbuildingId)

      stubTeleponyDeleteError(objectId = expectedRedirectObjectId)

      Get(s"/newbuilding/$newbuildingId/contacts") ~>
        route ~>
        check {
          status should be(StatusCodes.OK)
          val response = entityAs[NewbuildingContactsResponse]
          response.getResponse.getSalesDepartmentsCount === 1
        }

    }

  }

}

object NewbuildingApiSpec {

  def newbuildingRedirectObjectId(newbuildingId: Long): String = {
    s"site_$newbuildingId"
  }

}
