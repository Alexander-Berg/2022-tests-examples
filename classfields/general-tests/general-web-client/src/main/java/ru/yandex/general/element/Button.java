package ru.yandex.general.element;

import io.qameta.atlas.webdriver.extension.FindBy;
import io.qameta.atlas.webdriver.extension.Name;
import io.qameta.atlas.webdriver.extension.Param;
import ru.auto.tests.commons.extension.element.VertisElement;

public interface Button extends VertisElement {

    String ARIA_DISABLED = "aria-disabled";
    String TRUE = "true";
    String FALSE = "false";

    @Name("Кнопка «{{ value }}»")
    @FindBy(".//button[contains(.,'{{ value }}')]")
    VertisElement button(@Param("value") String value);

    @Name("Кнопка")
    @FindBy(".//button")
    VertisElement button();

}
