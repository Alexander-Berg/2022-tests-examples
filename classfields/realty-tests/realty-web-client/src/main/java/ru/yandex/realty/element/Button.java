package ru.yandex.realty.element;

import io.qameta.atlas.webdriver.AtlasWebElement;
import io.qameta.atlas.webdriver.extension.FindBy;
import io.qameta.atlas.webdriver.extension.Name;
import io.qameta.atlas.webdriver.extension.Param;

public interface Button extends AtlasWebElement {

    @Name("Кнопка «{{ value }}»")
    @FindBy(".//button[contains(.,'{{ value }}')]")
    RealtyElement button(@Param("value") String value);

    @Name("Кнопка «{{ value }}»")
    @FindBy(".//button[contains(.,'{{ value }}')]")
    RealtyElement buttonWithClickIf(@Param("value") String value);

    @Name("Кнопка")
    @FindBy(".//button")
    RealtyElement button();

}
