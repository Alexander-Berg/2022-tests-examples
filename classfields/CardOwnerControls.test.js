/**
 * @jest-environment @vertis/allure-report/build/circus/allure-jsdom-environment
 */

jest.mock('auto-core/react/dataDomain/card/actions/toggleAutoProlongation');
jest.mock('auto-core/react/dataDomain/card/actions/offerActivate');
jest.mock('auto-core/react/dataDomain/state/actions/paymentModalOpen');
jest.mock('auto-core/react/dataDomain/card/actions/updateOffer');

const React = require('react');
const { shallow } = require('enzyme');
const { shallowToJson } = require('enzyme-to-json');
const MockDate = require('mockdate');

const contextMock = require('autoru-frontend/mocks/contextMock').default;
const mockStore = require('autoru-frontend/mocks/mockStore').default;
const toggleAutoProlongation = require('auto-core/react/dataDomain/card/actions/toggleAutoProlongation');
const cardMock = require('auto-core/react/dataDomain/card/mocks/card.cars.mock');
const cloneOfferWithHelpers = require('autoru-frontend/mockData/state/helpers/offer/cloneOfferWithHelpers');
const { getBunkerMock } = require('autoru-frontend/mockData/state/bunker.mock');
const offerActivate = require('auto-core/react/dataDomain/card/actions/offerActivate');
const configStateMock = require('auto-core/react/dataDomain/config/mock').default;
const { SECOND } = require('auto-core/lib/consts');

const paymentModalOpen = require('auto-core/react/dataDomain/state/actions/paymentModalOpen');
paymentModalOpen.mockImplementation(() => () => {});

const updateOffer = require('auto-core/react/dataDomain/card/actions/updateOffer');
updateOffer.mockImplementation(() => () => {});

const CardOwnerControls = require('./CardOwnerControls');

let store;
beforeEach(() => {
    store = mockStore({
        bunker: getBunkerMock([ 'common/activate_in_app' ]),
        config: configStateMock.value(),
        user: { data: {} },
    });
});

afterEach(() => {
    MockDate.reset();
});

it('должен отрендерить контролы для владельца, активное объявление', () => {
    const offer = {
        additional_info: { is_owner: true, expire_date: 123 },
        actions: { edit: true, hide: true },
        category: 'cars',
        status: 'ACTIVE',
    };

    const wrapper = shallow(
        <CardOwnerControls offer={ offer }/>,
        { context: { ...contextMock, store } },
    ).dive().dive();

    expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it('не должен отрендерить контролы для невладельца', () => {
    const offer = {
        additional_info: { is_owner: false, expire_date: 123 },
        actions: { edit: true, hide: true },
        category: 'cars',
        status: 'ACTIVE',
    };

    const wrapper = shallow(
        <CardOwnerControls offer={ offer }/>,
        { context: { ...contextMock, store } },
    ).dive().dive();

    expect(shallowToJson(wrapper)).toHaveLength(0);
});

it('должен отрендерить доп инфо для перекупа с форсированным продлением размещения', () => {
    const offer = {
        additional_info: { is_owner: true, expire_date: '123' },
        actions: { edit: true, hide: true },
        category: 'cars',
        status: 'ACTIVE',
        service_prices: [ { service: 'all_sale_activate', prolongation_forced_not_togglable: true } ],
        services: [ { service: 'all_sale_activate', auto_prolong_price: 999 } ],
    };

    const wrapper = shallow(
        <CardOwnerControls offer={ offer }/>,
        { context: { ...contextMock, store } },
    ).dive().dive();

    expect(shallowToJson(wrapper.find('PlacementProlongationInfoDesktop'))).toMatchSnapshot();
});

it('должен отрендерить кнопку редактирования для дилера, если есть права на редактирование объявления', () => {
    const offer = {
        additional_info: { is_owner: true, expire_date: 123 },
        actions: { edit: true, hide: true },
        category: 'cars',
        status: 'ACTIVE',
        seller_type: 'COMMERCIAL',
    };

    store = mockStore({
        bunker: getBunkerMock([ 'common/activate_in_app' ]),
        config: configStateMock.value(),
        user: {
            data: {
                access: {
                    grants: [ {
                        access: 'READ_WRITE',
                        alias: 'OFFERS',
                    } ],
                },
                profile: {
                    autoru: {
                        client_id: '16453',
                    },
                },
            },
        },
    });

    const wrapper = shallow(
        <CardOwnerControls offer={ offer } canW/>,
        { context: { ...contextMock, store } },
    ).dive().dive();

    expect(wrapper.find('Link[children="Редактировать"]')).toHaveLength(1);
});

it('не должен отрендерить кнопку редактирования для дилера, если нет прав на редактирование объявления', () => {
    store = mockStore({
        bunker: getBunkerMock([ 'common/activate_in_app' ]),
        config: configStateMock.value(),
        user: {
            data: {
                access: {
                    grants: [ {
                        access: 'READ_ONLY',
                        alias: 'OFFERS',
                    } ],
                },
                profile: {
                    autoru: {
                        client_id: '16453',
                    },
                },
            },
        },
    });

    const offer = {
        additional_info: { is_owner: true, expire_date: 123 },
        actions: { edit: true, hide: true },
        category: 'cars',
        status: 'ACTIVE',
        seller_type: 'COMMERCIAL',
    };

    const wrapper = shallow(
        <CardOwnerControls offer={ offer } canW/>,
        { context: { ...contextMock, store } },
    ).dive().dive();

    expect(wrapper.find('Link[children="Редактировать"]')).toHaveLength(0);
});

it('должен быть статус "В архиве", удаленное объявление', () => {
    const offer = {
        additional_info: { is_owner: true, expire_date: 123 },
        actions: { edit: true, hide: true },
        category: 'cars',
        status: 'REMOVED',
    };

    const wrapper = shallow(
        <CardOwnerControls offer={ offer }/>,
        { context: { ...contextMock, store } },
    ).dive().dive();
    expect(shallowToJson(wrapper.find('div[children="В архиве"]'))).not.toBeNull();
});

it('Не должно быть кнопки "редактировать", если оффер не редактируется', () => {
    const offer = {
        additional_info: { is_owner: true, expire_date: 123 },
        actions: { edit: false, hide: true },
        category: 'cars',
        status: 'REMOVED',
    };

    const wrapper = shallow(
        <CardOwnerControls offer={ offer }/>,
        { context: { ...contextMock, store } },
    ).dive().dive();
    expect(shallowToJson(wrapper.find('Link[children="Редактировать"]'))).toBeNull();
});

it('должна быть кнопка "Опубликовать", если объявление можно активировать', () => {
    const offer = {
        additional_info: { is_owner: true, expire_date: 123 },
        actions: { edit: false, activate: true },
        category: 'cars',
        status: 'REMOVED',
    };

    const wrapper = shallow(
        <CardOwnerControls offer={ offer }/>,
        { context: { ...contextMock, store } },
    ).dive().dive();
    expect(shallowToJson(wrapper.find('Link[children="Опубликовать"]'))).not.toBeNull();
});

it('при клике на кнопку "Опубликовать" вызовет экшен для активации объявы', () => {
    offerActivate.mockImplementationOnce(() => () => Promise.resolve());
    const offer = {
        additional_info: { is_owner: true, expire_date: 123 },
        actions: { edit: false, activate: true },
        category: 'cars',
        status: 'REMOVED',
        saleId: '1085562758-1970f439',
    };

    const wrapper = shallow(
        <CardOwnerControls offer={ offer }/>,
        { context: { ...contextMock, store } },
    ).dive().dive();

    const publishLink = wrapper.find('Link[children="Опубликовать"]');
    publishLink.simulate('click');

    expect(offerActivate).toHaveBeenCalledTimes(1);
    expect(offerActivate).toHaveBeenCalledWith({ category: 'cars', from: 'card-vas', offerIdHash: '1085562758-1970f439' });
});

describe('при клике на кнопку "подключить автопродление размещения"', () => {
    const offer = {
        additional_info: { is_owner: true, expire_date: 123 },
        actions: { edit: true, hide: true },
        category: 'cars',
        status: 'ACTIVE',
    };

    beforeEach(() => {
        contextMock.metrika.sendParams.mockClear();
    });

    it('вызовет необходимый экшен', () => {
        toggleAutoProlongation.mockImplementationOnce(jest.fn(() => () => Promise.resolve()));
        const wrapper = shallow(
            <CardOwnerControls offer={ offer }/>,
            { context: { ...contextMock, store } },
        ).dive().dive();

        const placementAutoProlongationMessageBlock = wrapper.find('PlacementAutoProlongationInactiveNotice');
        placementAutoProlongationMessageBlock.simulate('prolongationButtonClick');

        expect(toggleAutoProlongation).toHaveBeenCalledTimes(1);
        expect(toggleAutoProlongation).toHaveBeenCalledWith(
            true,
            { product: 'all_sale_activate' },
            { exposeError: true },
        );
    });

    it('если действие было удачным отправит соответствующую метрику', () => {
        const pr = Promise.resolve();
        toggleAutoProlongation.mockImplementationOnce(jest.fn(() => () => pr));
        const wrapper = shallow(
            <CardOwnerControls offer={ offer }/>,
            { context: { ...contextMock, store } },
        ).dive().dive();

        const placementAutoProlongationMessageBlock = wrapper.find('PlacementAutoProlongationInactiveNotice');
        placementAutoProlongationMessageBlock.simulate('prolongationButtonClick');

        return pr
            .then(() => {
                expect(contextMock.metrika.sendParams).toHaveBeenCalledTimes(1);
                expect(contextMock.metrika.sendParams).toHaveBeenCalledWith([ '7days-placement', 'prolongation-turn-on', 'from-offer' ]);
            });
    });

    it('если действие было неудачным отправит соответствующую метрику', () => {
        const pr = Promise.reject();
        toggleAutoProlongation.mockImplementationOnce(jest.fn(() => () => pr));
        const wrapper = shallow(
            <CardOwnerControls offer={ offer }/>,
            { context: { ...contextMock, store } },
        ).dive().dive();

        const placementAutoProlongationMessageBlock = wrapper.find('PlacementAutoProlongationInactiveNotice');
        placementAutoProlongationMessageBlock.simulate('prolongationButtonClick');

        return pr.then(
            () => Promise.reject('UNEXPECTED_RESOLVE'),
            async() => {
                await new Promise((resolve) => setTimeout(resolve));
                expect(contextMock.metrika.sendParams).toHaveBeenCalledTimes(1);
                expect(contextMock.metrika.sendParams).toHaveBeenCalledWith([ '7days-placement', 'prolongation-errors' ]);
            },
        );
    });
});

describe('если у неактивного оффера есть продление со скидкой', () => {
    let wrapper;

    beforeEach(() => {
        const offer = cloneOfferWithHelpers(cardMock)
            .withAction({ activate: true })
            .withSellerTypePrivate()
            .withStatus('INACTIVE')
            .withCustomVas({
                service: 'all_sale_activate',
                price: 999,
                original_price: 1777,
                days: 7,
                prolongation_forced_not_togglable: true,
                prolongation_interval_will_expire: '2020-03-18T13:55:33Z',
            })
            .value();
        MockDate.set('2020-03-18T13:00:00Z');
        //@see https://github.com/facebook/jest/issues/11551
        jest.useFakeTimers('legacy');

        wrapper = shallow(
            <CardOwnerControls offer={ offer }/>,
            { context: { ...contextMock, store } },
        ).dive().dive();
    });

    it('при клике на кнопку откроет модал оплаты', () => {
        const discountBlock = wrapper.find('PlacementAutoProlongationExpireNotice');
        discountBlock.simulate('activateButtonClick', 'foo');

        expect(paymentModalOpen).toHaveBeenCalledTimes(1);
        expect(paymentModalOpen).toHaveBeenCalledWith('foo');
    });

    it('когда выйдет таймер обновит оффер', () => {
        const discountBlock = wrapper.find('PlacementAutoProlongationExpireNotice');
        discountBlock.simulate('timerFinish');

        expect(updateOffer).toHaveBeenCalledTimes(0);

        jest.advanceTimersByTime(3 * SECOND);

        expect(updateOffer).toHaveBeenCalledTimes(1);
        expect(updateOffer).toHaveBeenCalledWith({ category: 'cars', offerID: '1085562758-1970f439' });
    });
});
