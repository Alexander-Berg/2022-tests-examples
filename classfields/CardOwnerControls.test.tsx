/**
 * @jest-environment @vertis/allure-report/build/circus/allure-jsdom-environment
 */
jest.mock('auto-core/react/dataDomain/state/actions/paymentModalOpen');

import type { Action } from 'redux';
import React from 'react';
import type { ShallowWrapper } from 'enzyme';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { OfferStatus } from '@vertis/schema-registry/ts-types-snake/auto/api/api_offer_model';

import contextMock from 'autoru-frontend/mocks/contextMock';
import mockStore from 'autoru-frontend/mocks/mockStore';
import cloneOfferWithHelpers from 'autoru-frontend/mockData/state/helpers/offer/cloneOfferWithHelpers';
import hideMock from 'autoru-frontend/mockData/state/helpers/hideMock';

import offerMock from 'auto-core/react/dataDomain/card/mocks/card.cars.2.mock';
import Button from 'auto-core/react/components/islands/Button/Button';
import paymentModalOpen from 'auto-core/react/dataDomain/state/actions/paymentModalOpen';
import getServiceInfoMerged from 'auto-core/react/lib/offer/getServiceInfoMerged';

import { TOfferVas } from 'auto-core/types/proto/auto/api/api_offer_model';
import { TBillingFrom } from 'auto-core/types/TBilling';

import type { MobileAppState } from 'www-mobile/react/MobileAppState';

import CardOwnerControls from './CardOwnerControls';

const paymentModalOpenMock = paymentModalOpen as jest.MockedFunction<typeof paymentModalOpen>;
paymentModalOpenMock.mockReturnValue((() => () => { }) as unknown as Action);

let defaultProps: any;
let store: MobileAppState;
beforeEach(() => {
    defaultProps = {
        isPlacementDiscountActive: false,
        pageType: 'card',
        params: {
            from: TBillingFrom.MOBILE_CARD_PLACEMENT_PROLONG,
        },
        placementInfo: getServiceInfoMerged(offerMock, TOfferVas.PLACEMENT),
    };
    store = mockStore({
        bunker: {
            'common/activate_in_app': {
                marks: [],
            },
        },
        user: { data: {} },
    }) as Partial<MobileAppState> as MobileAppState;
});

describe('блок статистики', () => {
    it('должен отрисовать, если объявление частника активно', () => {
        const offer = cloneOfferWithHelpers(offerMock)
            .withSellerTypePrivate()
            .withStatus(OfferStatus.ACTIVE);

        const wrapper = shallow(
            <CardOwnerControls
                { ...defaultProps }
                offer={ offer.value() }
            />,
            { context: { store, ...contextMock } },
        ).dive().dive();

        expect(wrapper.find('.CardOwnerControlsMobile__stat')).toHaveLength(1);
    });

    it('не должен отрисовать, если объявление частника неактивно', () => {
        const offer = cloneOfferWithHelpers(offerMock)
            .withSellerTypePrivate()
            .withStatus(OfferStatus.INACTIVE);

        const wrapper = shallow(
            <CardOwnerControls
                { ...defaultProps }
                offer={ offer.value() }
            />,
            { context: { store, ...contextMock } },
        ).dive().dive();

        expect(wrapper.find('.CardOwnerControlsMobile__stat')).toHaveLength(0);
    });

    it('не должен отрисовать, если объявление дилера', () => {
        const offer = cloneOfferWithHelpers(offerMock)
            .withSellerTypeCommercial()
            .withStatus(OfferStatus.ACTIVE);

        const wrapper = shallow(
            <CardOwnerControls
                { ...defaultProps }
                offer={ offer.value() }
            />,
            { context: { store, ...contextMock } },
        ).dive().dive();

        expect(wrapper.find('.CardOwnerControlsMobile__stat')).toHaveLength(0);
    });

    it('должен отрисовать блок статистики без позиции в поиске, если ее нет', () => {
        const offer = cloneOfferWithHelpers(offerMock)
            .withSearchPosition(-1)
            .value();
        const wrapper = shallow(
            <CardOwnerControls
                { ...defaultProps }
                offer={ offer }
            />,
            { context: { store, ...contextMock } },
        ).dive().dive();
        const snapshot = shallowToJson(wrapper.find('.CardOwnerControlsMobile__stat'), { map: hideMock(offer, 'offer', '[Offer mock]') });
        expect(snapshot).toMatchSnapshot();
    });
});

describe('кнопки действий', () => {
    it('должен отрисовать правильный набор кнопок для редактирования и активации для частника', () => {
        const offer = cloneOfferWithHelpers(offerMock)
            .withSellerTypePrivate()
            .withAction({ edit: true, activate: true, hide: false, archive: true });

        const wrapper = shallow(
            <CardOwnerControls
                { ...defaultProps }
                offer={ offer.value() }
            />,
            { context: { store, ...contextMock } },
        ).dive().dive();

        expect(shallowToJson(wrapper.find('.CardOwnerControlsMobile__buttonGroup'))).toMatchSnapshot();
    });

    it('должен отрисовать только активацию для дилера', () => {
        const offer = cloneOfferWithHelpers(offerMock)
            .withSellerTypeCommercial()
            .withAction({ edit: true, activate: true, hide: false, archive: true });

        const wrapper = shallow(
            <CardOwnerControls
                { ...defaultProps }
                offer={ offer.value() }
            />,
            { context: { store, ...contextMock } },
        ).dive().dive();

        expect(shallowToJson(wrapper.find('.CardOwnerControlsMobile__buttonGroup'))).toMatchSnapshot();
    });

    it('должен отрисовать правильный набор кнопок для редактирования и снятия с продажи', () => {
        const offer = cloneOfferWithHelpers(offerMock)
            .withSellerTypePrivate()
            .withAction({ edit: true, activate: false, hide: true, archive: false });

        const wrapper = shallow(
            <CardOwnerControls
                { ...defaultProps }
                offer={ offer.value() }
            />,
            { context: { store, ...contextMock } },
        ).dive().dive();

        expect(shallowToJson(wrapper.find('.CardOwnerControlsMobile__buttonGroup'))).toMatchSnapshot();
    });

    it('должен отрисовать кнопку "написать в поддержку", если оффер забанен', () => {
        const offer = cloneOfferWithHelpers(offerMock)
            .withSellerTypePrivate()
            .withAction({ edit: false, activate: false, hide: false, archive: false })
            .withStatus(OfferStatus.BANNED);

        const wrapper = shallow(
            <CardOwnerControls
                { ...defaultProps }
                offer={ offer.value() }
            />,
            { context: { store, ...contextMock } },
        ).dive().dive();

        expect(shallowToJson(wrapper.find('.CardOwnerControlsMobile__buttonGroup'))).toMatchSnapshot();
    });
});

describe('если для оффера действует скидка на продление размещения', () => {
    let wrapper: ShallowWrapper;
    beforeEach(() => {
        const offer = cloneOfferWithHelpers(offerMock)
            .withAction({ edit: true, activate: true, hide: false, archive: true })
            .withStatus(OfferStatus.INACTIVE)
            .value();
        wrapper = shallow(
            <CardOwnerControls
                { ...defaultProps }
                offer={ offer }
                isPlacementDiscountActive
            />,
            { context: { store, ...contextMock } },
        ).dive().dive();
    });

    it('залогирует показ васа при маунте', () => {
        expect(contextMock.logVasEvent).toHaveBeenCalledTimes(1);
        expect(contextMock.logVasEvent.mock.calls[0]).toMatchSnapshot();
    });

    it('при клике на кнопку активировать вызовет модал оплаты', () => {
        wrapper.find('Button').find({ color: Button.COLOR.GREEN }).simulate('click');
        expect(paymentModalOpenMock).toHaveBeenCalledTimes(1);
        expect(paymentModalOpenMock.mock.calls[0]).toMatchSnapshot();
    });

    it('при клике на кнопку активировать залогирует клик', () => {
        wrapper.find('Button').find({ color: Button.COLOR.GREEN }).simulate('click');
        expect(contextMock.logVasEvent).toHaveBeenCalledTimes(2);
        expect(contextMock.logVasEvent.mock.calls[1]).toMatchSnapshot();
    });
});
