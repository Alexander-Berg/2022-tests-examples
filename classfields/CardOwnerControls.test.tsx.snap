// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`блок статистики должен отрисовать блок статистики без позиции в поиске, если ее нет 1`] = `
<div
  className="CardOwnerControlsMobile__stat"
>
  <div
    className="CardOwnerControlsMobile__statItem"
  >
    <div
      className="CardOwnerControlsMobile__statItemTitle"
    >
      Просмотры
    </div>
    <div
      className="CardOwnerControlsMobile__statItemCount"
    >
      11
    </div>
    <div
      className="CardOwnerControlsMobile__statItemTotal"
    >
      91
       всего
    </div>
  </div>
  <div
    className="CardOwnerControlsMobile__statItem"
  >
    <div
      className="CardOwnerControlsMobile__statItemTitle"
    >
      Звонки
    </div>
    <div
      className="CardOwnerControlsMobile__statItemCount"
    >
      3
    </div>
    <div
      className="CardOwnerControlsMobile__statItemTotal"
    >
      <Link
        onClick={[Function]}
      >
        12
         всего
      </Link>
      <OfferCallHistoryModal
        offer="[Offer mock]"
        onRequestHide={[Function]}
        visible={false}
      />
    </div>
  </div>
</div>
`;

exports[`если для оффера действует скидка на продление размещения залогирует показ васа при маунте 1`] = `
Array [
  Object {
    "category": "cars",
    "event": "VAS_SHOW",
    "from": "api_m_card_prolong",
    "offerId": "1085562758-1970f439",
    "originalPrice": undefined,
    "price": 2199,
    "serviceId": "all_sale_activate",
  },
]
`;

exports[`если для оффера действует скидка на продление размещения при клике на кнопку активировать вызовет модал оплаты 1`] = `
Array [
  Object {
    "category": "cars",
    "from": "api_m_card_prolong",
    "offerId": "1085562758-1970f439",
    "platform": "PLATFORM_MOBILE",
    "services": Array [
      Object {
        "service": "all_sale_activate",
      },
    ],
    "shouldShowSuccessTextAfter": true,
    "successText": "Платёж прошёл",
  },
]
`;

exports[`если для оффера действует скидка на продление размещения при клике на кнопку активировать залогирует клик 1`] = `
Array [
  Object {
    "category": "cars",
    "event": "VAS_CLICK",
    "from": "api_m_card_prolong",
    "offerId": "1085562758-1970f439",
    "originalPrice": undefined,
    "price": 2199,
    "serviceId": "all_sale_activate",
  },
]
`;

exports[`кнопки действий должен отрисовать кнопку "написать в поддержку", если оффер забанен 1`] = `
<div
  className="CardOwnerControlsMobile__buttonGroup"
>
  <Button
    className="CardOwnerControlsMobile__button"
    color="transparentBlue"
    onClick={[Function]}
    size="l"
    type="link"
    url="//auto.ru/?chat"
    width="full"
  >
    Написать в поддержку
  </Button>
</div>
`;

exports[`кнопки действий должен отрисовать правильный набор кнопок для редактирования и активации для частника 1`] = `
Array [
  <div
    className="CardOwnerControlsMobile__buttonGroup"
  >
    <Button
      className="CardOwnerControlsMobile__button"
      color="green"
      onClick={[Function]}
      size="l"
      type="button"
      width="full"
    >
      Активировать
    </Button>
  </div>,
  <div
    className="CardOwnerControlsMobile__buttonGroup"
  >
    <Button
      className="CardOwnerControlsMobile__button"
      color="transparentBlue"
      size="l"
      type="link"
      url="link/fromWebToApp/?action=edit"
      width="full"
    >
      Редактировать
    </Button>
    <Button
      className="CardOwnerControlsMobile__button"
      color="transparentBlue"
      onClick={[Function]}
      size="l"
      type="button"
      width="full"
    >
      Удалить
    </Button>
  </div>,
]
`;

exports[`кнопки действий должен отрисовать правильный набор кнопок для редактирования и снятия с продажи 1`] = `
<div
  className="CardOwnerControlsMobile__buttonGroup"
>
  <Button
    className="CardOwnerControlsMobile__button"
    color="transparentBlue"
    size="l"
    type="link"
    url="link/fromWebToApp/?action=edit"
    width="full"
  >
    Редактировать
  </Button>
  <Button
    className="CardOwnerControlsMobile__button"
    color="transparentBlue"
    onClick={[Function]}
    size="l"
    type="button"
    width="full"
  >
    Снять с продажи
  </Button>
</div>
`;

exports[`кнопки действий должен отрисовать только активацию для дилера 1`] = `
Array [
  <div
    className="CardOwnerControlsMobile__buttonGroup"
  >
    <Button
      className="CardOwnerControlsMobile__button"
      color="green"
      onClick={[Function]}
      size="l"
      type="button"
      width="full"
    >
      Активировать
    </Button>
  </div>,
  <div
    className="CardOwnerControlsMobile__buttonGroup"
  />,
]
`;
